contracts/layer-2/ChampMarketplace.sol

```diff
@@ -632,39 +679,48 @@ contract ChampMarketplace is AccessControlEnumerableUpgradeable {
             isReservationOpenFor(nftReceiver, tokenId),
             "ChampMarketplace: A reservation exists for this sale"
         );
-        require(
-            _CHAMP_TOKEN_CONTRACT.allowance(msg.sender, address(this)) >=
-                salePrice,
-            "ChampMarketplace: Allowance is lower than sale price"
-        );
 
         //
         // 2.
         // Process sale
         //
         address seller = _NFCHAMP_CONTRACT.ownerOf(tokenId);
-        uint256 sellerTokenWeiShare;
-        uint256 marketplaceFeesTokenWeiShare;
-        (sellerTokenWeiShare, marketplaceFeesTokenWeiShare) = computeSaleShares(
-            salePrice
+        (
+            uint256 sellerShare,
+            uint256 marketplaceSellFeeShare,
+            uint256 marketplaceBuyFeeShare,
+            uint256 marketplaceBurnFeeShare
+        ) = computeSaleShares(salePrice);
+
+        require(
+            _CHAMP_TOKEN_CONTRACT.allowance(msg.sender, address(this)) >=
+                salePrice + marketplaceBuyFeeShare,
```

Comment:
P4
```suggestion
                this.getBuyerSalePrice(tokenId),
```

Reply:
Here, we do `salePrice + marketplaceBuyFeeShare` specifically to avoid calling `getBuyerSalePrice`. The reason being this last function calls `computeSaleShares` internally (which is a relatively expensive function, due to all the storage reads and math calculations). 

`computeSaleShares` function is already being called in the code directly above this check. So, by having this explicit sum here instead of `getBuyerSalePrice`, we avoid having to call `computeSaleShares` function twice, reducing with some significance the gas cost of accepting a sale.